Load CMDB data when binaries start (before augments)

vpodzime opened the pull request:
Merge together:
https://github.com/cfengine/libntech/pull/119

vpodzime commented:
@cf-bottom jenkins, please

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=6226)](https://ci.cfengine.com//job/pr-pipeline/6226/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6226/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6226/

vpodzime commented:
@olehermanse I only introduced `JsonGetType()` in all the relevant new places and removed the unnecessary `stat()` stuff from `ReadJsonFile()`. I also rebased the branch so the force-push diff is big, though.

larsewi commented on libpromises/cmdb.c:
```diff
@@ -0,0 +1,246 @@
+        return false;
+    }
+
+    JsonElement *data = ReadJsonFile(file_path, LOG_LEVEL_ERR, HOST_SPECIFIC_DATA_MAX_SIZE);
```
I can only see json element `data` destroyed in case of error, on line 216. Should it not be destroyed before we return from this function no matter what?

vpodzime replied:
Good catch!

larsewi reviewed (commented):
Looking good! :+1:  I've left one comment on what may be a memory leak.
